src/_pytest/stepwise.py

```diff
@@ -65,26 +83,20 @@ def pytest_collection_modifyitems(
                 break
             else:
                 already_passed.append(item)
-
         # If the previously failed test was not found among the test items,
         # do not skip any tests.
         if not found:
-            self.report_status = "previously failed test not found, not skipping."
+            self.report_status = PREVIOUSLY_FAILED_TEST_NOT_FOUND
             already_passed = []
         else:
-            self.report_status = "skipping {} already passed items.".format(
-                len(already_passed)
-            )
+            self.report_status = SKIPPING_PASSED_ITEMS.format(len(already_passed))
 
         for item in already_passed:
```

Comment:
This of course is not part of your change, but just noticed his is `O(n)`... @symonk if you are willing, you can also save the index of the found item in the `for item in items` loop, and use simple slicing here to discard the items which already passed (no need to keep them on a list either).

The code even is easier to understand IMHO (this is applied on `master`):

```python
        passed_index = None
        for index, item in enumerate(items):
            if item.nodeid == self.lastfailed:
                passed_index = index
                break
            
        # If the previously failed test was not found among the test items,
        # do not skip any tests.
        if passed_index is None:
            self.report_status = "previously failed test not found, not skipping."
        else:
            self.report_status = f"skipping {passed_index} already passed items."

        items[:] = items[passed_index + 1:]
        if passed_index is not None:
            config.hook.pytest_deselected(items=already_passed)
```

Reply:
If you prefer to keep this PR small don't worry, I can open a separate PR with that small improvement above. 👍

Reply:
sure @nicoddemus I will look at enhancing the performance

Reply:
tho the code above doesn't quite look right with passing already_passed to the `deselected` hook

Reply:
Ahh right, my bad, fixed that and improved it further a little bit.